test/cql-pytest/test_permissions.py

```diff
@@ -39,3 +40,46 @@ def test_grant_applicable_data_and_role_permissions(cql, test_keyspace, cassandr
             for permission in all_permissions.difference(set(permissions)):
                 with pytest.raises((InvalidRequest, SyntaxException), match="support.*permissions"):
                     cql.execute(f"GRANT {permission} ON {resource} TO {user}")
+
+
+def eventually_authorized(fun, retries=100):
+    for i in range(retries):
+        try:
+            return fun()
+        except Unauthorized as e:
+            time.sleep(0.1)
+    return fun()
+
+def eventually_unauthorized(fun, retries=100):
+    for i in range(retries):
+        try:
+            fun()
+            time.sleep(0.1)
+        except Unauthorized as e:
+            return
+    try:
+        fun()
+        raise Exception(f"Function {fun} was not refused as unauthorized")
+    except Unauthorized as e:
+        return
+
+# Test that data permissions can be granted and revoked, and that they're effective
+# Unfortunately it takes ~3 seconds on Scylla and ~4 seconds on Cassandra to refresh
+# permissions, even though they're supposed to be refreshed on new login.
```

Comment:
Did we know about this refresh time? Do we consider this acceptable - I see you used the phrasing "even though", maybe it is a bug?

Anyway, as I noted above I think it's fine that we have a test that takes 3 seconds. It will be less fine to have 10 or 20 of them. The question is whether we could merge several tests into one. For example, grant to username the ability to do 7 unrelated things, and then with one sleep, check all those 7 unrelated things. I'm not sure we can actually merge all the different things into one test - and that it will be as powerful as the separate tests. But maybe it will be useful to have a least one test without the veryslow marker that can be run by Jenkins.

Reply:
This is a priceless comment!!!!!!!! I didn't even suspect that this period might be configurable and I was just looking for the hardcoded value in the code, but it looks like it's configurable after all:
https://github.com/scylladb/scylla/blob/7e2d2cf1c18ddf471df1c2a28e805a5c1c69af50/db/config.cc#L726-L727

, both in Scylla and Cassandra!

That's fantastic news, because it makes the whole `runveryslow` thing redundant. I verified locally and indeed the tests run in absolutely acceptable times when this variable is set to `100ms`.